### PR TITLE
Subscriber API gateway test coverage improvements

### PR DESCRIPTION
**Pull request by salman-codewar**
Issue: #1057
Includes:
- Minor refactoring related to unnecessary error handling
- Test coverage improved

----

> [!IMPORTANT]
> Improved test coverage and refactored error handling in Subscriber API gateway, enhancing robustness and maintainability.
> 
>   - **Test Coverage**:
>     - Added comprehensive test cases in `accounting_test.go`, `router_test.go`, and `event_test.go` for various scenarios including success, failure, and edge cases.
>     - Tests cover `Get`, `GetByUser`, `SyncAccounting`, and `EventNotification` functions.
>     - Added tests for handling invalid JSON, database errors, and message bus failures.
>   - **Refactoring**:
>     - Removed unnecessary error handling in `router.go` for functions like `getSimByIccid`, `getSims`, and `getSimPoolStats`.
>     - Simplified return statements by directly returning the result of client calls.
>   - **Helper Functions**:
>     - Introduced helper functions in `accounting_test.go` to generate test data for different scenarios.
>     - Added `handleEventCloudSimManagerSimAllocate` function in `event.go` for better modularity.
>   - **Miscellaneous**:
>     - Updated `utils.go` to include struct definitions for `Accounting`, `ConnectivityProvider`, `Nodes`, and `Item`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=ukama%2Fukama&utm_source=github&utm_medium=referral)<sup> for 64661c7c28f66adcd118148296ebb0f100cec307. You can [customize](https://app.ellipsis.dev/ukama/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: systems/subscriber/api-gateway/pkg/rest/router_test.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ import (
 	"github.com/gin-gonic/gin"
```
Comment (delvatt) [inline systems/subscriber/api-gateway/pkg/rest/router_test.go:22]: These commonly used dummy values:
```
                                        Iccid:          "1234567890123456789",
					SimType:        "ukama_data",
					Msisdn:         "555-555-1234",
					SmDpAddress:    "http://example.com",
					ActivationCode: "abc123",
					QrCode:         "qr123",
					and more ....
                                        and more ....
                                        and more ....
```
should be defined as **const** and reused throughout tests cases with minor variations, instead of having them hardcoded and duplicated overall in these tests cases.

**Review by delvatt — APPROVED** [review]
LGTM. 
Just some minor changes as stated earlier:

File: systems/subscriber/api-gateway/pkg/rest/router_test.go (hunk 1 above)
Comment (salman-codewar) [reply systems/subscriber/api-gateway/pkg/rest/router_test.go:22]: addressed

**Review by Brackleycassinga — APPROVED** [review]
